snapcraft/internal/lxd.py

```diff
@@ -139,6 +144,54 @@ def _setup_project(self):
         self._push_file(tar_filename, dst)
         self._container_run(['tar', 'xvf', os.path.basename(tar_filename)])
 
+    def _inject_snapcraft(self):
+        if is_snap():
+            # Because of https://bugs.launchpad.net/snappy/+bug/1628289
+            self._container_run(['apt-get', 'install', 'squashfuse', '-y'])
+
+            # Push core snap into container
+            self._inject_snap('core')
+            self._inject_snap('snapcraft')
+        else:
+            self._container_run(['apt-get', 'install', 'snapcraft', '-y'])
+
+    def _inject_snap(self, name):
+        session = requests_unixsocket.Session()
+        snapd_socket = '/run/snapd.socket'.replace('/', '%2F')
+        # Cf. https://github.com/snapcore/snapd/wiki/REST-API#get-v2snapsname
+        api = 'http+unix://{}/v2/snaps/{}'.format(snapd_socket, name)
+        try:
+            json = session.request('GET', api).json()
+        except Exception as e:
+            raise SnapcraftEnvironmentError(
+                'Error connecting to {}'.format(api)) from e
+        if json['type'] == 'error':
+            raise SnapcraftEnvironmentError(
+                'Error querying {!r} snap: {}'.format(
+                    name, json['result']['message']))
+        classic = json['result']['confinement'] == 'classic'
+        rev = json['result']['revision']
```

Comment:
Also make it track the same `channel` as on the system (maybe not as it won't be easier to switch channels from local installs in the same transaction as it is when installing from the store).